types/src/traits/node_implementation.rs

```diff
@@ -54,6 +55,200 @@ pub trait NodeImplementation<TYPES: NodeType>: Send + Sync + Debug + Clone + 'st
+        network_id: usize,
+    ) -> Box<
+        dyn Fn(
+                u64,
+            ) -> <Self::CommitteeExchange as ConsensusExchange<
+                TYPES,
+                Self::Leaf,
+                Message<TYPES, Self>,
+            >>::Networking
+            + 'static,
+    >;
+
+    fn quorum_generator(
+        expected_node_count: usize,
+        num_bootstrap: usize,
+        network_id: usize,
+    ) -> Box<
+        dyn Fn(
+                u64,
+            ) -> <Self::QuorumExchange as ConsensusExchange<
+                TYPES,
+                Self::Leaf,
+                Message<TYPES, Self>,
+            >>::Networking
+            + 'static,
+    >;
+
+    /// Get the number of messages in-flight from quorum exchange.
+    ///
+    /// Some implementations will not be able to tell how many messages there are in-flight. These implementations should return `None`.
+    fn quorum_in_flight_message_count(
+        network: &<Self::QuorumExchange as ConsensusExchange<
+            TYPES,
+            Self::Leaf,
+            Message<TYPES, Self>,
+        >>::Networking,
+    ) -> Option<usize>;
+
+    /// Get the number of messages in-flight from committee exchange.
+    ///
+    /// Some implementations will not be able to tell how many messages there are in-flight. These implementations should return `None`.
+    fn committee_in_flight_message_count(
+        network: &<Self::CommitteeExchange as ConsensusExchange<
+            TYPES,
+            Self::Leaf,
+            Message<TYPES, Self>,
+        >>::Networking,
+    ) -> Option<usize>;
+
+    /// Creates random transaction if possible
+    /// otherwise panics
+    /// `padding` is the bytes of padding to add to the transaction
+    fn state_create_random_transaction(
+        state: Option<&TYPES::StateType>,
+        rng: &mut dyn rand::RngCore,
+        padding: u64,
+    ) -> <TYPES::BlockType as Block>::Transaction;
+
+    fn leaf_create_random_transaction(
+        leaf: &Self::Leaf,
+        rng: &mut dyn rand::RngCore,
+        padding: u64,
+    ) -> <TYPES::BlockType as Block>::Transaction;
+
+    /// generate a genesis block
+    fn block_genesis() -> TYPES::BlockType;
+
+    fn txn_count(block: &TYPES::BlockType) -> u64;
+
+    // Create ephemeral storage
+    // Will be deleted/lost immediately after storage is dropped
+    // # Errors
+    // Errors if it is not possible to construct temporary storage.
+    fn construct_tmp_storage() -> Result<Self::Storage, StorageError>;
+
+    // Return the full internal state. This is useful for debugging.
+    async fn get_full_state(storage: &Self::Storage) -> StorageState<TYPES, Self::Leaf>;
+
+    fn generate_test_key(id: u64) -> <TYPES::SignatureKey as SignatureKey>::PrivateKey;
+
+    // /// Generate a vote token used for testing.
+    // fn quorum_generate_test_vote_token() -> TYPES::VoteTokenType;
+
+    // fn committee_generate_test_vote_token() -> TYPES::VoteTokenType;
```

Comment:
Do we plan to add this later? Or could we remove it here and from the `impl` below?

Reply:
Ah, that's dead code whoops 👍